lib/_http_incoming.js

```diff
@@ -63,6 +60,16 @@ util.inherits(IncomingMessage, Stream.Readable);
 
 exports.IncomingMessage = IncomingMessage;
 
+Object.defineProperty(IncomingMessage.prototype, 'client', {
+  configurable: true,
+  enumerable: true,
+  get: util.deprecate(function() {
+    return this.socket;
+  }, 'client is deprecated, use socket or connection instead'),
+  set: util.deprecate(function(val) {
+    this.socket = this.connection = val;
+  }, 'client is deprecated, use socket or connection instead')
```

Comment:
(As written this changes behavior – previously setting `.client` would not affect `.socket`.)